### ISSUE TITLE
Support rectangular selection within a block

### ISSUE THREAD

**Issue by pcholakov** [warpdotdev/Warp#2165]
### Describe the solution you'd like?

```text
Is it possible to perform rectangular text selection in Warp? If so, I haven't figured out how to do so. In iTerm2, holding down Cmd + Alt while dragging causes the selection mode to change to rectangular. VSCode calls this "column selection".
```

### Is your feature request related to a problem? Please describe.

```text
Having rectangular selection makes it much easier to copy fixed layout output or prefixed text.
```

### Additional context

https://code.visualstudio.com/docs/editor/codebasics#_column-box-selection
https://www.jetbrains.com/help/rider/Multicursor.html#column_selection

### How important is this feature to you?

4

**Comment by pcholakov** [comment]
This is what it looks like in iTerm2:

<img width="578" alt="image" src="https://user-images.githubusercontent.com/115803/202476585-d441e25c-3fb8-4b17-a260-aa9a25c463ca.png">

**Comment by dannyneira** [comment]
Thanks for this feature request! Please +1 this request as it helps us gauge interest. We will post any updates here.

**Comment by fty1777** [comment]
This feature would be a real lifesaver in some log reading and output capturing circumstances.

**Comment by dportalesr** [comment]
Absolutely.

**Comment by manewitz** [comment]
+1 for me. In other programs, I've heavily used Command-Option drag for rectangle/column selection (to copy items out of a REPL/console session without the prompt for instance)

**Comment by itshanine** [comment]
I installed warp earlier and switched back to iterm2 when i saw that they don't have this feature, guess i'll have to wait for it

**Comment by mikenerone** [comment]
Yes please. I like a lot about Warp, and this is one of the three or four things that are painful to me.

**Comment by nevdelap** [comment]
This is the thing that has forced me back to Alacritty.

**Comment by augustin4s** [comment]
The copy on select, which was recently fixed, was one of the two main missing features for me. The rectangular select feature is the last one holding me back from fully migrating to Warp

**Comment by mikenerone** [comment]
Any movement on this? This is pretty much a standard feature in other terminal apps. iTerm, alacritty, KDE Konsole, Gnome Terminal, PuTTY, heck even Windows Terminal, all do this. I'm having more trouble thinking of a major terminal app that _doesn't_ provide this. I like Warp a lot and I can wait for this a little while longer, but I just want to make it clear that this is a several-times-a-day source of frustration.

**Comment by GraphBear** [comment]
This is likely going to be the final "little thing" that sends me—albeit reluctantly—back to iTerm2.

It's obviously a big enough deal to enough people that there have been several tickets opened for it over the years.  And it's one of those "expected" features that even the lowly Apple Terminal has, let alone the many other terminals and IDEs already mentioned.

Can Warp please at least share when this is likely to be implemented?

**Comment by runarorama** [comment]
Need dis

**Comment by MostHated** [comment]
This is a pretty important feature for any terminal. The lack of it often makes me have to open up alacritty and then I end up just continuing on using it instead.

**Comment by bitwalk** [comment]
Is this still not implemented?

**Comment by kevinyang372** [comment]
Hey y'all! This feature was just released on the latest version (`v0.2025.01.23.08.02.dev_00`)

<img width="1181" alt="Image" src="https://github.com/user-attachments/assets/c97c19cc-f89f-4f70-8d6d-ae613dbef644" />

**Comment by mikenerone** [comment]
For anyone trying to figure out _how_ to use this, it's not in the docs nor is there any reference in Settings. The only place I've seen it mentioned is in the [changelog](https://docs.warp.dev/getting-started/changelog#id-2025.01.22-v0.2025.01.22.08.02):
> Added support for rectangular selection when holding ⌘⌥ on Mac and Ctrl-Alt on Linux.

**Comment by augustin4s** [comment]
Just a note, in vim buffer the rectangular select does not work, but this might be a separate ticket material
